agent/agent.go

```diff
@@ -550,12 +550,19 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 	detail := a.endTaskResponse(tc, status, message)
 	switch detail.Status {
 	case evergreen.TaskSucceeded:
-		tc.logger.Task().Info("Task completed - SUCCESS.")
-		a.runPostTaskCommands(ctx, tc)
+		if err := a.runPostTaskCommands(ctx, tc); err != nil {
```

Comment:
I think the issue with this is that most users won't be using post_error_fails_task, but now they suddenly have to wait for post task to know if the task has been successful or not. Maybe the more consistent user experience would be to log this before post task still, and then log "Post task completed -- FAILURE" if we're allowing that (this may be mildly confusing for that use case but it'll be more rare anyway)